mypy/checker.py

```diff
@@ -67,7 +67,32 @@ def min_with_None_large(x: T, y: T) -> T:
 
 
 class Frame(Dict[Any, Type]):
-    pass
+    """Frame for the ConditionalTypeBinder.
+
+    Frames actually in the binder must have
+    binder.frames[i].parent == binder.frames[i-1].
+
+    options_on_return represents a list of possible frames that we
+    will incorporate next time we are the top frame (e.g. frames from
+    the end of each if/elif/else block).
+    """
+
+    changed = False
+
+    def __init__(self, parent: 'Frame' = None) -> None:
+        super().__init__()
+        self.parent = parent
+        self.options_on_return = []  # type: List[Frame]
```

Comment:
Would it be correct to say that there are really two types involved here, a `FrameData` (or whatever) that just contains the expression->type mapping data, and a subclass (or even composite object?) `FrameInBinder` that has a `parent`, `options_on_return`, ...? I'm left wondering what parts of the `Frame`s stored in `options_on_return` will actually be used later; I suspect it's not much.